src/Classes/PolygonMask.ts

```diff
@@ -0,0 +1,247 @@
+                case window.SVGPathSeg.PATHSEG_ARC_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_ABS:
+                    const absSeg = segment as SvgAbsoluteCoordinatesTypes;
+
+                    p.x = absSeg.x;
+                    p.y = absSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_ABS:
+                    p.x = (segment as SVGPathSegLinetoHorizontalAbs).x;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_ABS:
+                    p.y = (segment as SVGPathSegLinetoVerticalAbs).y;
+                    break;
+
+                //
+                // Relative
+                //
+                case window.SVGPathSeg.PATHSEG_LINETO_REL:
+                case window.SVGPathSeg.PATHSEG_MOVETO_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_REL:
+                case window.SVGPathSeg.PATHSEG_ARC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_REL:
+                    const relSeg = segment as SvgRelativeCoordinatesTypes;
+
+                    p.x += relSeg.x;
+                    p.y += relSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_REL:
+                    p.x += (segment as SVGPathSegLinetoHorizontalRel).x;
+                    break;
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_REL:
+                    p.y += (segment as SVGPathSegLinetoVerticalRel).y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_UNKNOWN:
+                case window.SVGPathSeg.PATHSEG_CLOSEPATH:
+                    continue; // Skip the closing path (and the UNKNOWN)
+            }
+
+            polygonRaw.push([p.x * scale + this.offset.x, p.y * scale + this.offset.y]);
+        }
+
+        return polygonRaw;
+    }
+
+    public drawPolygon(): void {
+        const container = this.container;
+        const options = container.options;
+        const context = container.canvas.context;
+
+        if (context && this.raw) {
+            context.beginPath();
+            context.moveTo(this.raw[0][0], this.raw[0][1]);
+
+            for (let i = 1; i < this.raw.length; i++) {
+                context.lineTo(this.raw[i][0], this.raw[i][1]);
+            }
+
+            context.closePath();
+            context.strokeStyle = options.polygon.draw.lineColor;
+            context.lineWidth = options.polygon.draw.lineWidth;
+            context.stroke();
+        }
+    }
+
+    public drawPointsOnPolygonPath(): void {
+        const container = this.container;
+
+        if (this.raw) {
+            for (let i = 0; i < this.raw.length; i++) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected a 'for-of' loop instead of a 'for' loop with this simple iteration](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5052388)